### PR TITLE
add manual batch transactions

### PR DESCRIPTION
**Pull request by kordano**
#### SUMMARY
Implements #403 

##### Feature
- [x] Implements an existing feature request. Make sure the feature request has been accepted for implementation before opening a PR.
- [x] Related issues linked using `fixes #number`
- [x] Integration tests added
- [ ] Documentation added
- [x] Architecture Decision Record added 
- [x] Formatting checked

### CODE REVIEW

**Comment by whilo** [comment]
Just a minor remark, while `sync?` makes some sense as a flag, it will clash with the naming of the `core.async` flag we have now in konserve and might be confusing. Probably we should either fully namespace-qualify both or rename one.

**Review by jsmassa — DISMISSED** [review]
Nice job! I like that you also added specs :)

File: src/datahike/api.cljc [hunk 1]
```diff
@@ -12,19 +14,34 @@
...
           `:store` defines the backend configuration as hash-map with mandatory key: `:backend` and store dependent keys.
 
           Per default Datahike ships with `:mem` and `:file` backend.
 
-          The default configuration:
+          The default store configuration:
            `{:store {:backend :mem :id \"default\"}}`
 
+          Optionally a connection configuration hash-map with following keys can be added:
+          - `:tx/sync?`: boolean that whether transactions are flushed automatically on `transact`, if false the user needs to manually flush the database via `sync!`
```
Comment (jsmassa) [inline src/datahike/api.cljc:38]: word missing?

Comment (kordano) [reply src/datahike/api.cljc:38]: I'll have to rebase anyway and fix that with it.

**Pull request closed without merging**